.github/workflows/check.yml

```diff
@@ -0,0 +1,46 @@
+# Based upon:
+#   https://github.com/actions/starter-workflows/blob/main/ci/go.yml
+
+on:
+  pull_request:
+  push:
+    branches: [ $default-branch ]
+
+jobs:
+  check:
+
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+    strategy:
+      matrix:
+        go-version: [1.13, 1.14, 1.15]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: 'recursive'
+
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Download Packages
+        run: go get -t -v ./ably/...
+
+      - name: Vet
+        run: go vet ./ably ./scripts
```

Comment:
Maybe we can reuse commands from makefile. Possibly need to update some of the makefile commands to keep both in sync
```make vet``` 
should work in this case

Reply:
I've deleted the `Makefile`. It was unnecessary.

Reply:
If it's not needed anymore, then good

Reply:
I'm not sure I agree with replacing `make test` with `for both JSON and MessagePack protocols (see [workflow](.github/workflows/check.yml) for the test commands executed in the CI environment)` as the way to run the tests, why make developers jump through that hoop, especially casual contributors?